### PR TITLE
Cfddem dynamic contact search

### PR DESCRIPTION
**Pull request by shahabgol**
This PR adds dynamic contact search to unresolved CFD-DEM solver.

### CODE REVIEW

**Comment by blaisb** [comment]
@tonielgeitani can you review first I'll review afterwards.

File: include/fem-dem/cfd_dem_coupling.h [hunk 1]
```diff
@@ -152,7 +163,7 @@ class CFDDEMSolver : public GLSVANSSolver<dim>
 
 
   unsigned int                coupling_frequency;
-  bool                        contact_detection_step = false;
+  bool                        contact_detection_step = true;
```
Comment (tonielgeitani) [inline include/fem-dem/cfd_dem_coupling.h:166]: I don't understand why contact_detection_step should be initially set to true.

**Review by tonielgeitani — APPROVED** [review]

**Comment by tonielgeitani** [comment]
@blaisb I have worked on this branch today and added the fluidized_bed test. So maybe @shahabgol  can take a look at the test and check if its good.

File: applications_tests/cfd_dem_coupling_3d/CMakeLists.txt [hunk 2]
```diff
@@ -10,11 +10,19 @@ file(COPY particle_sedimentation_files/dem.triangulation DESTINATION ${CMAKE_CUR
...
+file(COPY fluidized_bed_files/dem.triangulation.info DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/fluidized_bed.${CMAKE_LOWER_BUILD_TYPE})
+file(COPY fluidized_bed_files/dem.triangulation_fixed.data DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/fluidized_bed.${CMAKE_LOWER_BUILD_TYPE})
+file(COPY fluidized_bed_files/dem.triangulation_variable.data DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/fluidized_bed.${CMAKE_LOWER_BUILD_TYPE})
 
 
 DEAL_II_PICKUP_TESTS()
 
 if (CMAKE_BUILD_TYPE STREQUAL "Debug" )
        	set_tests_properties(cfd_dem_coupling_3d/particle_sedimentation.debug PROPERTIES TIMEOUT 1200) 
+       	set_tests_properties(cfd_dem_coupling_3d/fluidized_bed.debug PROPERTIES TIMEOUT 2000)
```
Comment (blaisb) [inline applications_tests/cfd_dem_coupling_3d/CMakeLists.txt:26]: It's always a concern when tests are becoming this long. Is it possible to design a shorter one?

File: include/fem-dem/cfd_dem_coupling.h (hunk 1 above)
Comment (blaisb) [reply include/fem-dem/cfd_dem_coupling.h:166]: Don't give default value in the .h. If you want to give it a value, give it in the constructor....

File: source/fem-dem/cfd_dem_coupling.cc [hunk 3]
```diff
@@ -156,11 +172,32 @@ CFDDEMSolver<dim>::read_dem()
...
+  const unsigned int &counter)
 {
   return ((counter % contact_detection_frequency) == 0);
 }
 
+template <int dim>
+inline bool
+CFDDEMSolver<dim>::check_contact_search_step_dynamic(const unsigned int &)
+{
+  bool sorting_in_subdomains_step =
```
Comment (blaisb) [inline source/fem-dem/cfd_dem_coupling.cc:185]: Add comments or documentation here, this is not clear

**Review by blaisb — COMMENTED** [review]
Some comments to address before this can be merged.

**Comment by blaisb** [comment]
The test should be made much smaller.
The dem.triangulation_variable.data  is 81 MB! We cannot keep such a large file in the repository. You need to shrink the test in size AND duration.

File: source/fem-dem/cfd_dem_coupling.cc (hunk 3 above)
Comment (tonielgeitani) [reply source/fem-dem/cfd_dem_coupling.cc:193]: done

**Comment by tonielgeitani** [comment]
I changed the test and made it much smaller. I reduced the test timeout to 1200 s in debug model. All comments are now addressed.

**Comment by blaisb** [comment]
@tonielgeitani Can you rename the CFD-DEM example. Each example should be in it's own sub folder but, more importantly, it is not a good idea to put spaces in folder name. This is a bit a recipe for disaster :). Juste rename it to square_fluidized_bed or something similar :)

**Review by blaisb — COMMENTED** [review]
This is good for merge. Just rename the example to have no spaces and no capital letter.
Also, there is a merge conflict? Can you rebase / merge and fix the conflict. This way I will be able to merge.

**Review by blaisb — APPROVED** [review]

**Pull request merged**